classes/class-wc-rest-connect-base-controller.php

```diff
@@ -63,14 +71,44 @@ public function get_internal( $request ) {
 		if ( ! defined( 'DONOTCACHEPAGE' ) ) {
 			define( 'DONOTCACHEPAGE', true ); // Play nice with WP-Super-Cache
 		}
-		return $this->get( $request );
+		try {
+			return $this->get( $request );
+		} catch ( WCC_Api_Exception $error ) {
+			return $error->response;
+		}
 	}
 
 	public function post_internal( $request ) {
 		if ( ! defined( 'DONOTCACHEPAGE' ) ) {
 			define( 'DONOTCACHEPAGE', true ); // Play nice with WP-Super-Cache
 		}
-		return $this->post( $request );
+		try {
+			return $this->post( $request );
+		} catch ( WC_Connect_Api_Exception $ex ) {
+			return $ex->response;
+		}
+	}
+
+	protected function api_request( $method /*, ...$args */ ) {
+		$args = array_slice( func_get_args(), 1 );
+		$response = call_user_func_array( array( $this->api_client, $method ), $args );
+
+		if ( is_wp_error( $response ) ) {
+			$error = new WP_Error(
+				$response->get_error_code(),
+				$response->get_error_message(),
+				array( 'message' => $response->get_error_message() )
+			);
+			$this->logger->debug( $error, get_class( $this ) );
+			throw new WC_Connect_Api_Exception( $error );
+		}
+
+		if ( isset( $response->async_token ) || isset( $response->async_status ) ) {
+			// Not an error per-se, but we want to interrupt further processing of the response at this point
```

Comment:
> What do you think about refactoring other REST controllers to use api_request() so they can benefit from the built in WP_Error handling? It looks like a good opportunity to remove some redundant code.

That was the intention, but it would add more noise to this PR.

> We can leave the refactor for later work, which gives us time to rethink the exception approach as well.

👍